effects.c

```diff
@@ -134,6 +134,33 @@ static void effect_greyscale(uint32_t *data, int width, int height) {
 	}
 }
 
+static void effect_vignette(uint32_t *data, int width, int height,
+		double base, double factor) {
+	base = fmin(1, fmax(0, base));
```

Comment:
Not sure about the style here, does it make sense to put it under the `#pragma` below?

Reply:
That looks about like how I'd do it. It makes sense to me to have the `#pragma omp parallel for` right before the for loop it's parallelizing.